### ISSUE TITLE
Bring `apitools` Python3 support.

### ISSUE THREAD

**Issue by dhermes** [google/apitools#1]
See @craigcitro [comment][1].

PS Craig, https://github.com/google/protorpc does not have issues turned on.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/pull/654#issuecomment-76243464

**Comment by craigcitro** [comment]
@dhermes thanks for the heads-up! fixed.

**Comment by dhermes** [comment]
Filed there too. /cc @tseaver

**Comment by craigcitro** [comment]
quick question for @tseaver: when you applied the `python-modernize` changes to this repo, did you use any special flags?

**Comment by tseaver** [comment]
I don't recall any flags (the fixer I wanted to disable was adding `from __future__ import absolute_imports`, but I couldn't figure out how).

My memory is that I generated a patch file and edited it to remove that import and correct for some other wonkiness.

**Comment by craigcitro** [comment]
OK, we're now **much** closer: https://pypi.python.org/pypi/google-apitools

i've also done a massive update from internal, and merged in the `gsutil` fork. we're unblocked on better py3 support, IIUC.

my next step is to move `gsutil` over to using this code as a submodule. along those lines, i noticed that one of the changes in 901342e seemed to cause some unicode-related issues when i ran some (still-only-internal) tests; since it's late, i just stripped it out for now, but i think that one might require some digging.

**Comment by dhermes** [comment]
w00t

**Comment by craigcitro** [comment]
OK, i made a small effort here in 3c522a4 ... but there's still hard work to be done.

in particular, i need a plan to update or drop `gflags` and `apputils`. ugh.

**Comment by dhermes** [comment]
Drop, drop, drop, drop! Woo hoo!

Make them optional in `setup.py` and then everyone is happy. You can go back to the origin `old_run.py` in `oauth2client` to see how we ditched `gflags` there. (I think you were part of those code reviews way back when.)

**Comment by dhermes** [comment]
@craigcitro Can this be closed? Are `gflags` and `apputils` dropped?

**Comment by craigcitro** [comment]
So `gflags` and `apputils` aren't gone, but i at least have a plan in place. given what you did in #5, i'm gonna close this, and open a new issue to completely drop those dependencies.
